Modal column with comments

gizzmojr opened the pull request:
Split up the modal layout to be in columns. Following the styling queues of Github issue tracker, comments on the left and metadata on the right.

leif81 committed:
Put bug metadata in modal column

Trying to separate bug description and comments from metadata.

gizzmojr committed:
Prevent the additional comment textarea to push modal elements unexpectedly

gizzmojr committed:
Show all comments

gizzmojr committed:
Restoring the ability to add comment on resolution change
Broke on previous commit

gizzmojr committed:
Add some spinner feedback while we return comments

gizzmojr commented:
Thinking of refactoring createSpinner, so I can add one to the comment section of the modal body. 

Seems hidden when in the background as is.
Thoughts?

leif81 commented:
Ya a spinner may help. But maybe there is also an underlying issue? The response times on our internal bz site seem slow. It's not apples and apples, but doing a comments query on a Mozilla bug is as much as 7x faster than instance. 300ms vs ~2000ms. At 300ms we could get away without a spinner. 😎 

https://bugzilla.mozilla.org/rest/bug/9998/comment
http://bugzilla.msec.local/rest/bug/14951/comment

Anything we can do server side to speed up our bz instance?

leif81 committed:
refactor comments loop

Remove things the description field has in common with comments.
Keep things DRY.
Side effect of 'add comment' box at the bottom now, but not a problem.

gizzmojr commented:
Can you try non-FQDN for me, http://bugzilla/rest/bug/14951/comment
Mindful that Mozilla has proper web servers, load balancer, etc. Probably runs off the cloud too. 

Thinking the spinner still is a good idea while the page is doing work and something is displayed (modal).

Not a fan of having the comment box at the bottom thou.I I originally formatted it to be similar to the native BZ interface. Having to scroll down takes away from the speediness we're trying to maintain I feel.

leif81 committed:
Rename element id bug-comment

More fitting now.

leif81 committed:
Hide modal body until response comes back

Feels less awkward than showing partial modal.

leif81 committed:
Tweak modal column sizing

And moves scroll bar to modal body which feels bit more natural.

gizzmojr commented:
I like the hiding until ready, that feels more fitting and expected behavior

leif81 commented:
Ya changing it to bottom posting was a bit of an accident. But now that I'm conscious of it I see github and google inbox do that too :thinking:. It's more scrolling, but seeing what's being replied to for context is handy and seems more intuitive doesn't it? I'm tempted to change my bz user pref now!

leif81 commented:
@gizzmojr I tried the non-fqdn link, it's ~1s. Then I tried the original fqdn link and it's the same ballpark. Which is better than what I original was seeing, but still ~3x slower than the Mozilla link.

I think what we've got in this branch is good enough to merge, unless there's anything left you wanted to tackle?

gizzmojr committed:
Fixing column width for all conditions

gizzmojr commented:
Ok now it looks good, found an issue with column widths for new bugs and when in resolved state.

leif81 reviewed (commented):
Hmm this seems to break the case when a comment contains a really long line. The comment line previously would wrap. Not sure of the fix yet other than reverting. What was this one fixing?

gizzmojr commented:
If the card action was anything other then click (show comments) the styling was all wrong. Meta width be 2/3 of the modal, description and/or add comment was tiny, etc. 
Your class (bug-comment) was doing all the work but wasn't for all cases. 

I see your issue. It's one continuous word is the issue, not a sentence. I'll investigate.

gizzmojr committed:
Fixes long character words in comments
EM unit type might not be what we want

leif81 committed:
Fix broken layout in login modal

Ensure our new column styling is only applied in the bug modal.

leif81 commented on bzkanban.js:
```diff
@@ -1362,6 +1362,7 @@ function showBugModal(bugCurrent, bugUpdate) {
     meta.className = 'bug-meta';
 
     body.appendChild(comments);
+    showSpinner();
```
How come?

gizzmojr replied:
Can be reverted, it was related to the discussion about slowness and I was adding some type of user notification. https://github.com/leif81/bzkanban/pull/57#issuecomment-276247231

In fact, could be switched out for a spinning cursor

leif81 replied:
Ah Ok, was just the position of it that threw me off. I think it'd be better to do it before the `httpGet` in that function bc I don't think we need a spinner for the "else" case (i.e. when a card is dragged).

gizzmojr replied:
Thoughts on making it a cursor instead?

https://css-tricks.com/almanac/properties/c/cursor/ (progress or wait cursor)

leif81 replied:
Maybe ya. For now I added a commit to move the spinner logic in there.

leif81 committed:
Show spinner only when modal is doing a get

leif81 commented:
I'm seeing one other case where the `bug-comment` div isn't right. If the content is less than our desired width it shrinks to that.

leif81 committed:
Set min width for comment div

gizzmojr committed:
Set max width for comment div
Feel there's a better solution somehow
